### PR TITLE
[25705] Refactor BigQuery SchemaTransforms naming

### LINKED ISSUE #25705: [Task]: Rename BigQuery SchemaTransform providers
### What needs to happen?

The naming convention for Schema transform providers should follow:
`<Resource>(Read|Write)SchemaTransformProvider`.

Additionally, the identifier should follow:
`beam:schematransform:org.apache.beam:<resource>_(read|write):<version>`.

Finally, each SchemaTransformProvider should be annotated with the `@AutoService(SchemaProvider.class)` annotation.

Therefore, this task should:
- [x] Rename BigQuerySchemaTransform(Read|Write)Provider to BigQuery(Read|Write)SchemaTransformProvider
- [x] Rename the configuration classes
- [x] Change the identifiers
- [x] Add the `@AutoService(SchemaProvider.class)` annotation

### Issue Priority

Priority: 2 (default / most normal work should be filed as P2)

### Issue Components

- [ ] Component: Python SDK
- [X] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by damondouglas**
This PR closes #25705 by
- Rename provider classes and configuration
- Rename identifiers
- Add the `@AutoService(SchemaProvider.class)` annotation

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 ~Update `CHANGES.md` with noteworthy changes.~
 ~If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).~

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by ahmedabu98** [comment]
The read method here uses the [`EXPORT`](https://github.com/apache/beam/blob/39cab94361bc44e8b48257385d37d8978ab7fc02/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java#L864) read method. There's a a `DIRECT` read schematransform here ([BigQueryDirectReadSchemaTransformProvider](https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryDirectReadSchemaTransformProvider.java)), maybe this should also have a more precise name and include "Export"?

Similarly with the write schematransform, it uses the FILE_LOADS method. In comparison, there is the `STORAGE_WRITE_API` schematransform here ([BigQueryStorageWriteApiSchemaTransformProvider](https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryStorageWriteApiSchemaTransformProvider.java))

P.S. I know this makes the names super long, so I'm open to a better way of doing this.

**Comment by ahmedabu98** [comment]
Also, there's a `../gcp/bigquery/providers` now, can you move these under there?

**Comment by damondouglas** [comment]
Run Java_GCP_IO_Direct PreCommit

**Comment by damondouglas** [comment]
> Also, there's a `../gcp/bigquery/providers` directory now, can you move these files under there?

I like the idea of organizing into a single folder providers.  However, I looked into this but many of the tests used BigQueryIO related testing utilities that would require that I change their modifiers to public.

**Comment by damondouglas** [comment]
R: ahmedabu98

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryFileLoadsSchemaTransformConfiguration.java [hunk 1]
```diff
@@ -27,25 +27,25 @@
...
+ * <p>This class is meant to be used with {@link BigQueryFileLoadsWriteSchemaTransformProvider}.
  *
  * <p><b>Internal only:</b> This class is actively being worked on, and it will likely change. We
  * provide no backwards compatibility guarantees, and it should not be implemented outside the Beam
  * repository.
  */
 @DefaultSchema(AutoValueSchema.class)
 @AutoValue
-public abstract class BigQueryWriteSchemaTransformConfiguration {
+public abstract class BigQueryFileLoadsSchemaTransformConfiguration {
```
Comment (ahmedabu98) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryFileLoadsSchemaTransformConfiguration.java:38]: wdyt of `BigQueryFileLoadsWriteSchemaTransformConfiguration`? better to keep provider and configuration name consistent

**Review by ahmedabu98 — CHANGES_REQUESTED** [review]

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryExportSchemaTransformConfiguration.java [hunk 2]
```diff
@@ -28,7 +28,7 @@
 /**
  * Configuration for reading from BigQuery.
  *
- * <p>This class is meant to be used with {@link BigQueryReadSchemaTransformProvider}.
+ * <p>This class is meant to be used with {@link BigQueryExportSchemaTransformProvider}.
```
[the remark this answers is not in the document]
Comment (damondouglas) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryExportSchemaTransformConfiguration.java:31]: Naming is hard and I really like your ideas.

**Comment by damondouglas** [comment]
Run Java_GCP_IO_Direct PreCommit

**Comment by damondouglas** [comment]
Run Java_Examples_Dataflow_Java11 PreCommit

**Comment by ahmedabu98** [comment]
LGTM

**Comment by Abacn** [comment]
Ideally shouldn't different read or write methods just being a field of the schema transform? Or if they're so different that different provider are needed. I'm fine with this change as it clean up existing namings. Merging for now

**Pull request merged**